### PR TITLE
Removed restangular from FederationService

### PR DESCRIPTION
**Pull request by ocket8888**
## What does this PR (Pull Request) do?
Removes a dependency on Restangular from the "FederationService"
- [x] This PR partially addresses #3571 

## Which Traffic Control components are affected by this PR?

- Traffic Portal

Traffic Portal dependencies are not individually documented, and so no documentation changes are necessary.

## What is the best way to verify this PR?

No functionality should have changed (except that errors will now be logged instead of ignored in many cases), so the existing tests should all pass.

Jeremy: actually, the only way to truly verify that the functionality did not change is to:

1. run the UI tests (which is a very small subset of all TP functionality)
2. manually verify the functionality associated with ALL the services functions that changed just to be sure all is well.

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY**

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3796/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/4023/
Test PASSed.

File: traffic_portal/app/src/common/api/FederationService.js [hunk 1]
```diff
@@ -17,136 +17,178 @@
...
 	};
 
 	this.getFederationDeliveryServices = function(fedId) {
-		return Restangular.one('federations', fedId).getList('deliveryservices');
+		return $http.get(ENV.api['root'] + 'federations/' + fedId + '/deliveryservices').then(
+			function (result) {
+				return result.data.response;
+			},
+			function (err) {
+				return throw err;
```
Comment (mitchell852) [inline traffic_portal/app/src/common/api/FederationService.js]: remove the return

File: experimental/traffic-portal-angular7/package.json [hunk 2]
```diff
@@ -32,6 +32,7 @@
     "chart.js": "^2.8.0",
     "core-js": "^2.5.4",
     "express": "^4.15.2",
+    "node-sass": "^4.13.0",
```
Comment (mitchell852) [inline experimental/traffic-portal-angular7/package.json]: looks like a couple of things slipped into this commit @ocket8888

Comment (ocket8888) [reply experimental/traffic-portal-angular7/package.json]: whoops. I don't have my ZSH extensions installed yet, so I have a hard time telling what branch I'm on sometimes

**Pull request merged**